src/main/scala/is/hail/stats/RegressionUtils.scala

```diff
@@ -51,39 +52,46 @@ object RegressionUtils {
     new SparseVector[Double](rows.result(), valsArray, row)
   }
 
-  // mask == null is interpreted as no mask
-  // impute uses mean for missing dosage rather than Double.NaN
-  def dosages(gs: Iterable[Genotype], nKept: Int, mask: Array[Boolean] = null, impute: Boolean = true): DenseVector[Double] = {
+  def dosages(X: DenseVector[Double], gs: Iterable[Genotype], completeSampleIndex: Array[Int], missingSamples: ArrayBuilder[Int]) {
```

Comment:
Since it's duplicated a few times, maybe we should maintain a shim:

```scala
def dosages(gs: Iterable[Genotype], completeSampleIndex: Array[Int]): DenseVector[Double] = {
  val x = new DenseVector[Double](n)
  val missingSamples = new ArrayBuilder[Int]()
  RegressionUtils.dosages(x, gs, completeSampleIndexBc.value, missingSamples)
  x
}
```